### ISSUE TITLE
Z Loss in CE

### ISSUE THREAD

**Issue by Fr0do** [linkedin/Liger-Kernel#197]
### 🚀 The feature, motivation and pitch

Often used in pretraining of LMs for stabilization, i.e. the recent [Chameleon](https://arxiv.org/abs/2405.09818) & [PaLM](https://www.jmlr.org/papers/v24/22-1144.html).

### Alternatives

[flash-attn](https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/ops/triton/cross_entropy.py) has implementations of abovementioned features, however, does not support fusing with linear head.

**Comment by ByronHsu** [comment]
Legit ask! We have tracked smooth label at https://github.com/linkedin/Liger-Kernel/issues/81. I modify the title for only Z loss to prevent duplication.

**Comment by Tcc0403** [comment]
@ByronHsu #take 
To support z loss, I just need a little add-ons to #198. 
I'll work on it after merging label_smoothing PR.
